subprojects/packagefiles/google-brotli/meson.build

```diff
@@ -26,18 +26,19 @@ project(
   meson_version: '>= 0.54',
 )
 
+c_args = []
+
 cc = meson.get_compiler('c')
 
 dep_m = cc.find_library(
   'm',
   required: false,
 )
-if not cc.has_function(
+has_log2 = cc.has_function(
```

Comment:
I personally use

```
 if cc.links('#include <math.h>\nint main(){log(0);}')
   m_dep = declare_dependency()
 else
   m_dep = cc.find_library('m')
 endif
```

to avoid a pointless dependency on libm. It's a moot point though.

Reply:
I find it hard to believe a system supports meson but not log2.

Reply:
I find it very easy to believe that a system doesn't support meson natively, but does allow cross compiling TO that system from another system which does support meson.

And that the build machine running meson does have log2, but the host machine perhaps doesn't have log2.

Reply:
I'm not arguing that something is sensible, I'm arguing that for ports we should not derivative from the official build system except when:
  1. Meson provides different way to achieve the same thing (sanitizers, static vs shared vs both, etc)
  2. is for some platform we don't support
  3. cannot be done
  4. there is a bug in the upstream build system

Maybe this case is moot since Brotli 1.2 was released recently and it has dropped this option.